### PR TITLE
API access with API key + update docs

### PR DESCRIPTION
**Pull request by Shr1ftyy**

### CODE REVIEW

File: docs/validator.md [hunk 1]
```diff
@@ -20,6 +20,61 @@ dbmate --url "sqlite:validator_database.db" up
...
+
+```bash
+storb create-key --balance 10 --rate-limit-per-minute 60 --name test
+```
+Creates a test key with a balance of 10 (which corresponds to 10 requests), a rate limit of 60 requests per minute = 1/s, and a name 'test'.
+
+<!-- **Recommend values:**
+- Balance: Depends on how much you want to sell! Each credit is a image (so a balance of 1000 will allow 1000 images to be generated)
+- Rate limit: I would recommend a rate limit of ~20/minute for casual users trying out the API, and around ~120/minute for production users
+- Name: Just for you to remember who you want to use that key :) -->
```
Comment (rishiad) [inline docs/validator.md:55]: reword this to fit our subnet

File: neurons/validator.py [hunk 2]
```diff
@@ -273,6 +276,61 @@ async def start_dht(self) -> None:
...
+            content={"detail": "API key is missing"},
+        )
+
+    async with db.get_db_connection(db_dir=core_validator.config.db_dir) as conn:
+        api_key_info = await db.get_api_key_info(conn, api_key)
+
+    if api_key_info is None:
+        return JSONResponse(status_code=HTTP_401_UNAUTHORIZED, content={"detail": "Invalid API key"})
+
+    credits_required = 1  # TODO: make this non-constant in the future???? (i.e. dependent on number of pools)????
```
Comment (rishiad) [inline neurons/validator.py:307]: Put this in the constants file

Comment (rishiad) [reply neurons/validator.py:307]: also the `credits_required` should be dependent on the size of the file. The biggest cost for the validator will be bandwidth

**Review by rishiad — CHANGES_REQUESTED** [review]
The economics of this needs to be hashed out a bit more

File: docs/validator.md (hunk 1 above)
Comment (Shr1ftyy) [reply docs/validator.md:55]: im just gonna remove it - the params are pretty self-explanatory anyway.

**Comment by rishiad** [comment]
We have decided to not to build the api monetisation right now, we will visit this idea later.
